eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartPaymentFSM.scala

```diff
@@ -98,6 +104,7 @@ class MultiPartPaymentFSM(nodeParams: NodeParams, paymentHash: ByteVector32, tot
         case PaymentSucceeded(parts) =>
           // We expect the parent actor to send us a PoisonPill after receiving this message.
           parent ! MultiPartHtlcSucceeded(paymentHash, parts)
+          Metrics.ReceivedPaymentDuration.withTag(Tags.Success, value = true).record(Platform.currentTime - start, TimeUnit.MILLISECONDS)
```

Comment:
This could be factored in another `onTransition`, not sure if worth it.

Reply:
I think it would use more code without adding much value, I'll leave it unchanged for now.